server/api/cart.js

```diff
@@ -0,0 +1,66 @@
+const router = require('express').Router();
+const User = require("../db/models/User");
+const Order = require("../db/models/Order");
+// const Category = require("../db/models/Category");
+// const LineItem = require("../db/models/Lineitem");
+// const Product = require("../db/models/Product");
+
+// GET /api/cart
+router.get("/", async (req, res, next) => {
+  //find user by token and get their cart
+  try {
+    const user = await User.findByToken(req.headers.authorization);
+    res.send(await user.getCart());
+  } catch (error) {
+    next(error);
+  }
+});
+
+// POST /api/addToCart
+router.post("/addToCart", async (req, res, next) => {
+  try {
+    const user = await User.findByToken(req.headers.authorization);
+    res.send(await user.addToCart(req.body)); //req.body = product
+  } catch (error) {
+    next(error);
+  }
+});
+
+// POST /api/removeFromCart
+router.post("/removeFromCart", async (req, res, next) => {
```

Comment:
Similarly, this could make sense as a DELETE route to `/api/cart`, since removing something from the cart actually destroys an entry in the database